packages/backend/src/api/controllers/ForcedActionsController.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  renderNewPerpetualForcedTradePage,
+  renderNewSpotForcedWithdrawPage,
+} from '@explorer/frontend'
+import { UserDetails } from '@explorer/shared'
+import { AssetHash, AssetId, EthereumAddress } from '@explorer/types'
+
+import { UserService } from '../../core/UserService'
+import { AssetRepository } from '../../peripherals/database/AssetRepository'
+import { PreprocessedAssetHistoryRepository } from '../../peripherals/database/PreprocessedAssetHistoryRepository'
+import { ControllerResult } from './ControllerResult'
+
+export class ForcedActionsController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly preprocessedAssetHistoryRepository: PreprocessedAssetHistoryRepository<
+      AssetHash | AssetId
+    >,
+    private readonly assetRepository: AssetRepository
+  ) {}
+
+  async getSpotForcedWithdrawalPage(
+    givenUser: Partial<UserDetails>,
+    vaultId: bigint
+  ): Promise<ControllerResult> {
+    const user = await this.userService.getUserDetails(givenUser)
+
+    if (!user) {
+      return { type: 'not found', content: 'User must be logged in' }
+    }
+
+    const assets =
+      await this.preprocessedAssetHistoryRepository.getCurrentByPositionOrVaultId(
+        vaultId
+      )
+
+    const asset = assets[0]
+    if (!asset) {
+      return { type: 'not found', content: 'Vault is empty' }
+    }
+
+    if (asset.starkKey != user.starkKey) {
+      return { type: 'not found', content: 'Vault does not belong to user' }
+    }
+
+    const assetDetails = await this.assetRepository.findDetailsByAssetHash(
+      asset.assetHashOrId as AssetHash
+    )
+
+    const content = renderNewSpotForcedWithdrawPage({
+      user,
+      starkExAddress: EthereumAddress.ZERO,
```

Comment:
I assume this is the address of the StarkEx contract. It's in the config, although it's incorrectly named `contracts.perpetual` even for spot. Please pass the correct address here and  rename the config field (or create a story to do it later). I suggest renaming `contracts.perpetual` to `contracts.starkex` for both spot and perpetual.

Reply:
I kinda feel that contracts.starkex is not the thing to go for because the full path would go like this `config.starkex.contracts.starkex`. Maybe we can have a call to talk about this later?